src/spikeinterface/core/waveform_tools.py

```diff
@@ -397,6 +398,288 @@ def _waveform_extractor_chunk(segment_index, start_frame, end_frame, worker_ctx)
 
 
+def extract_waveforms_to_single_buffer(
+    recording,
+    spikes,
+    unit_ids,
+    nbefore,
+    nafter,
+    mode="memmap",
+    return_scaled=False,
+    folder=None,
+    dtype=None,
+    sparsity_mask=None,
+    copy=False,
+    job_name=None,
+    **job_kwargs,
+):
+    """
+    Allocate a single buffer (memmap or or shared memory) and then distribute every waveform into it.
+
+    Contrary to extract_waveforms_to_buffers() all waveforms are extracted in the same buffer, so the spike vector is
+    needed to recover waveforms unit by unit. Importantly in case of sparsity, the channel are not aligned across
+    units.
+
+    Important note: for the "shared_memory" mode wf_array_info contains reference to
+    the shared memmory buffer, this variable must be reference as long as arrays as used.
+    And this variable is also returned.
+    To avoid this a copy to non shared memmory can be perform at the end.
+
+    Parameters
+    ----------
+    recording: recording
+        The recording object
+    spikes: 1d numpy array with several fields
+        Spikes handled as a unique vector.
+        This vector can be obtained with: `spikes = Sorting.to_spike_vector()`
+    unit_ids: list ot numpy
+        List of unit_ids
+    nbefore: int
+        N samples before spike
+    nafter: int
+        N samples after spike
+    mode: str
+        Mode to use ('memmap' | 'shared_memory')
+    return_scaled: bool
+        Scale traces before exporting to buffer or not.
+    folder: str or path
+        In case of memmap mode, folder to save npy files
+    dtype: numpy.dtype
+        dtype for waveforms buffer
+    sparsity_mask: None or array of bool
+        If not None shape must be must be (len(unit_ids), len(channel_ids))
+    copy: bool
+        If True (default), the output shared memory object is copied to a numpy standard array.
+        If copy=False then arrays_info is also return. Please keep in mind that arrays_info
+        need to be referenced as long as waveforms_by_units will be used otherwise it will be very hard to debug.
+        Also when copy=False the SharedMemory will need to be unlink manually
+    {}
+
+    Returns
+    -------
+    all_waveforms: numpy array
+        Single array with shape (nump_spikes, num_samples, num_channels)
+
+    wf_array_info: dict of info
+        Optionally return in case of shared_memory if copy=False.
+        Dictionary to "construct" array in workers process (memmap file or sharemem info)
+    """
+    nsamples = nbefore + nafter
+
+    dtype = np.dtype(dtype)
+    if mode == "shared_memory":
+        assert folder is None
+    else:
+        folder = Path(folder)
+
+    num_spikes = spikes.size
+    if sparsity_mask is None:
+        num_chans = recording.get_num_channels()
+    else:
+        num_chans = max(np.sum(sparsity_mask, axis=1))
+    shape = (num_spikes, nsamples, num_chans)
+
+    if mode == "memmap":
+        filename = str(folder / f"all_waveforms.npy")
```

Comment:
```suggestion
        filename = str(folder / f"waveforms.npy")
```
I would call it just `waveforms` because there is no guarantee that all spikes are used. No?

Reply:
I think we should even give the filename in the signature instead of give folder no ?

Reply:
Finally I remove the folder kwargs and put file_path instead which is even better the name is external.
We will need this in several context.
I think this is better. Tell me.